### DISCUSSION TITLE
Pairing Weaviate with Dify for RAG

### DISCUSSION THREAD

**Discussion by EdgarBabajanyan** [langgenius/dify#4442] [Help]
Hey everyone! I have a vector database in a platform called Weaviate and I want to use Dify as a front-end to interact with that database for a RAG project. Below are a few of the links saying that I can work with Weaviate on the Dify platform

https://docs.dify.ai/getting-started/install-self-hosted/environments
https://docs.dify.ai/tutorials/vector-database-migrate-tool

Now my confusion gets to the workflow where I see only knowledge retrieval done with Dify and no option to select Weaviate as my knowledge retrieval provider
I went into my docker-compose and adjusted the weaviate credentials and ran the flask db migrate command with a successful migration message.

What am I doing wrong? I still see no option to integrate a Weaviate Cluster into my project and for additional information, this cluster will eventually be stored on aws, for now it is stored on the official weaviate cloud services page (so I know the endpoint is 100% okay)

![image-36.png](https://github.com/langgenius/dify/assets/50426742/d93588f1-4dee-4375-8ba8-b30986b5db89)

![image-25.png](https://github.com/langgenius/dify/assets/50426742/c7e6ff8c-c30f-4204-a367-3f259d3568fd)

![image-20.png](https://github.com/langgenius/dify/assets/50426742/3ab4a02a-d043-4dbc-828e-223168dc285e)

**Comment by patryk20120** [comment]
Well, the migration tutorial is about migrating already existing data in vector DB that Dify uses e.g. when you change services from weaviate to qdrant - not importing external DB. 

Dify already comes with Weaviate DB. You can try to import the data directly into Weaviate
